include/seastar/net/tcp.hh

```diff
@@ -2090,7 +2088,7 @@ tcp_seq tcp<InetTraits>::tcb::get_isn() {
     hash[1] = _foreign_ip.ip;
     hash[2] = (_local_port << 16) + _foreign_port;
     hash[3] = _isn_secret.key[15];
-    CryptoPP::Weak::MD5::Transform(hash, _isn_secret.key);
+    gnutls_hash_fast(GNUTLS_DIG_MD5, _isn_secret.key, sizeof(_isn_secret.key), &hash);
```

Comment:
I'm not an expert in these functions, but a quick glance on the gnutls_hash_fast() suggests that the last parameter is where the output is stored. So what's point of setting it in advanced?
I see below you have another use of gnutls_hash_fast() which doesn't set the output array...

Reply:
see https://www.rfc-editor.org/rfc/rfc6528#section-3, we are generating the Initial Sequence Number Generation (ISN for short here), the md5 hash serves as the part of the function of `F` in 

```
ISN = M + F(localip, localport, remoteip, remoteport, secretkey)
```

where "F() is a pseudorandom function (PRF) of the connection-id."

to be more specific,  we just use the first byte of the md5 digest for `F(localip, localport, remoteip, remoteport, secretkey)`, as it fulfill the needs of the `F()` described in the RFC. the confusion comes from the somewhat unusual use of hash -- before my change, we used it as both input parameter and output parameter, `CryptoPP::Weak::MD5::Transform()` calculate the hash incrementally, so all 512 bits of `_isn_secret.key` are used without copying. see https://cryptopp.com/docs/ref/md5_8cpp_source.html#l00043 .

but i failed to take this into consideration in my previous change. and that's wrong. so in the latest revision, a hash handle is used for calculating the hash in multiple passes. and i put the rationales in the commit message.